Set/Set/SetGame.swift

```diff
@@ -7,37 +7,53 @@
 
 import Foundation
 
+enum CardState {
+    case chosen, matched, unmatched
+}
+
 final class SetGame {
 
     var canDrawCards: Bool {
-        self.shownCards.count < 24
+        (self.shownCards.compactMap { $0 }.count < 24 ||
+         self.matchedCardsIndices.count > 2) &&
+        self.deck.count > 2
     }
+
     private var deck = Array<Card>()
-    private var shownCards: Array<Card>
+    private var shownCards: Array<Card?>
     private var chosenCardsIndices: Array<Int>
     private var matchedCardsIndices: Array<Int>
+    private var unmatchedCardsIndices: Array<Int>
+    private(set) var score: Int
 
     init() {
-        self.shownCards = []
+        self.shownCards = .init(repeating: nil, count: 24)
         self.chosenCardsIndices = []
         self.matchedCardsIndices = []
+        self.unmatchedCardsIndices = []
+        self.score = 0
         generateDeck()
     }
 
-    func drawCards(_ amount: Int) {
-        for _ in 0..<amount {
-            drawOneCard()
+    func dealCards(_ amount: Int) {
```

Comment:
Let's try to do logic comparison between your and mine implementation. Both version seems working but I still wondering why we have this difference 🤔 
<img width="416" alt="image" src="https://github.com/khaletska/Set/assets/13601748/53cf8b30-1a8e-4ca1-b20f-faaf26608df6">

Reply:
I can see at least one case that you are not covering